### PR TITLE
OP-34: Hypothesis property tests — scale and translation invariance

### PR DESCRIPTION
**Pull request by m-nweke**
This pull request introduces comprehensive property-based tests to the `posture-core` package to rigorously verify the correctness and robustness of the redesigned posture analysis engine. The new tests use the `hypothesis` library to ensure that the engine's metrics are invariant to body size, image frame size, and subject position, and that the report generation is robust against missing or low-confidence data. Additionally, `hypothesis` is added as a development dependency.

**Testing improvements and verification of invariants:**

* Added `test_properties.py` with property-based tests that prove the engine's angular metrics are invariant to body size, image frame size, and subject translation, using `hypothesis` to generate a wide range of test cases. This directly verifies that the redesign fixed the original scale-dependence defect.
* Added property-based tests to guarantee that the `build_report` function never raises exceptions, even with missing or low-confidence landmarks, and always produces a deterministic report for the same input.

**Dependency updates:**

* Added the `hypothesis` library to the development dependencies in `pyproject.toml` to enable property-based testing.

### CODE REVIEW

File: packages/posture-core/tests/test_properties.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    }
+
+    baseline = angular_values(original)
+    moved = angular_values(shifted)
+    for name, value in baseline.items():
+        other = moved[name]
+        if value is not None and other is not None:
+            assert other == pytest.approx(value, abs=1e-6), f"{name} moved under translation"
+
+
```
[the remark this answers is not in the document]
Comment (m-nweke) [reply packages/posture-core/tests/test_properties.py:232]: Agreed, and what makes it clearly an oversight rather than a decision is that the two sibling properties directly above — scale invariance and frame-size invariance — both already assert `(value is None) == (other is None)` before comparing. This loop was the only one of the three exempting itself.

Added the same assertion here. A translation that changes whether a metric can be computed at all is a larger regression than one that changes its value, and it was precisely the case being skipped.

Also applied the same guard to `test_the_view_ratio_is_invariant_to_body_size` further down, which had the identical gap for the same reason.

Both still pass over 200 examples, so this is closing a hole rather than papering over a failure.

File: packages/posture-core/tests/test_properties.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+            x=landmark.x,
+            y=landmark.y,
+            visibility=landmark.visibility,
+            presence=landmark.presence,
+            x_world=None if landmark.x_world is None else landmark.x_world + dx,
+            y_world=None if landmark.y_world is None else landmark.y_world + dy,
+            z_world=landmark.z_world,
+        )
+        for name, landmark in original.items()
+    }
```
[the remark this answers is not in the document]
Comment (m-nweke) [reply packages/posture-core/tests/test_properties.py]: Correct — the docstring said "every world coordinate" and the code shifted two of three. Now generating a 3-tuple and shifting `z_world` as well.

I checked which metrics this actually reaches rather than assuming, and the answer is interesting enough to have put in the comment:

- `craniovertebral_angle_deg` — perturbing `z_world` alone moves it from 50.0° to 66.3°
- `knee_flexion_deg` — 100.0° to 96.4°
- `trunk_inclination_deg` — unmoved, and **correctly so**. Once the facing direction is resolved, depth is that metric's *lateral* axis, so a sideways displacement should not change a forward lean.

So the gap was real for two of the four angular metrics. My first version of the comment guessed that trunk inclination would be the depth-sensitive one, which the measurement disproved; the comment now records what was measured.

Still passing over 200 examples with all three axes shifted, which is the substantive result: translation invariance holds in 3D, not just in the image plane.

File: packages/posture-core/tests/test_properties.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+        assert scaled == pytest.approx(baseline, abs=1e-6)
+
+
+# ---------------------------------------------------------------------------------------------
+# Nothing raises, ever
+# ---------------------------------------------------------------------------------------------
+
+
+@SETTINGS
+@given(pose=poses, dropped=st.sets(st.sampled_from(list(K)), max_size=34))
```
[the remark this answers is not in the document]
Comment (m-nweke) [reply packages/posture-core/tests/test_properties.py]: Agreed. Now `KEYPOINT_COUNT = len(K)`, used by both strategies.

The two hard-coded sites would also have failed differently, which is worth noting for whoever adds keypoint 35. The `visibilities` strategy zips against every name with `strict=True`, so a stale literal there fails loudly. The `dropped` strategy's `max_size` does not — it would quietly stop being able to drop every landmark, and the "nothing raises however much is missing" property would still pass while no longer testing the empty-frame case it exists for. Silent under-coverage in the test that guards the legacy engine's worst failure mode.
